### DISCUSSION TITLE
Font Design and Accessibility

### DISCUSSION THREAD

**Discussion by Myndex** [w3c/low-vision-a11y-tf#116] [General]
In the CSS forum, I had made an offhand comment about Microsoft Core Web Fonts implying they are mostly not accessible (the notable exceptions being Georgia and Verdana). Someone that works at Microsoft was defensive and [so I responded further](https://github.com/w3c/csswg-drafts/issues/5065#issuecomment-929869890) with some examples. Some of the topic is useful for discussion, and illustrates some of the accessibly problems with fonts today.

## _Accessible Font Design_
Let me pause for a moment here: I am not talking about fonts as far as their aesthetic design. I am talking strictly about _accessibility_, including research on dyslexia, cognitive, contrast perception, and visual impairments. 

### Some accessibility points are: does a given font...
- have adequate tracking, kerning, and leading? _(letter spacing and linespacing)_
- have an adequate x-height? _(Ideally ~60% of the font-size)_
- have adequate weight? _(Adequate stroke thickness at 400)_
- have a "good" Cap-height? _(For differentiating from x and allowing good leading)_
- avoid common homoglyphs? _(Meaning_ **rn** _vs_ **m** and **1** _vs_ **l** _vs_ **I** _are distinct.)_
- avoid mirror image glyphs? _(Meaning_ **db** or **qp** or **MW** _are not literal mirror copies)_
- have good hinting for screen use? _(Meaning is designed to render well to standard resolution screens.)_

Most of these characteristics are discussed prominently in the current research into readability. For cites, I refer to Lovie-Kitchin, Bailey, Arditi, Legge all of whom I rely on and who have done the bulk of the work in readability for normal and impaired vision. All that said:

### Re: MS Fonts
Microsoft does have some fonts that are fairly accessibly. For instance Sitka Text was a Matthew Carter creation like Verdana and Georgia, [but created in a program with empirical research](https://www.researchgate.net/publication/309754147_Sitka_A_collaboration_between_type_design_and_science#fullTextFileContent) so not surprising it's a fairly strong font and fairly accessible.

As a result of the other thread, I did a much deeper dive into the history of "MS Fonts" by which I mean the "core web fonts" which MS distributed once upon a time. Those were the proprietary fonts _Andalé Mono, Arial, Arial Black, Comic Sans MS, Courier New, Georgia, Impact, Times New Roman, Trebuchet MS, Verdana_ and _Webdings,_

Of _THAT_ group, only Georgia, Verdana, and _maybe_ Trebuchet MS are "accessible friendly". The problem though with Trebuchet is the default tracking/kerning is too tight, so it's not normally on my list of "good" fonts for accessibility.

But also of that group, _Courier New_ is nothing short of an **_unusable illegitimate version of the classic Courier typeface._** _Arial_ is nothing but a pure copy of _Helvetica_, which I also give somewhat low marks to regarding _accessibility_. _Times New Roman_ with its small size and small x-height make it less than desirable for web use. The small size of Times made it a choice for newspapers who were concerned with jamming as much text into as small a space as possible to maximize use of paper. 

As a result of the deeper dive, I see that _some_ of the fonts I loath the most were Microsoft commissions to Monotype. ..... And looking further, Monotype is behind a few other fonts I consider "bad" which have nothing to do with Microsoft, and a lot of Monotype fonts are just knockoffs of some other foundry's font. Microsoft seems to have a close business relationship to Monotype.

Nevertheless, it was MS that _chose_ the core fonts, and Microsoft has more recent fonts that are also poor for accessibility. Though in fairness, most foundries do, considering that accessibility had not been at the forefront of design until the last few decades. (As a civil rights movement, accessibility is among the newest, gaining public awareness in the 1970s.)

But getting back to _Courier New_ it is not even a faithful copy and it's horrendous for screen use. Examples:

<img width="681" alt="Font Size Compared" src="https://user-images.githubusercontent.com/42009457/135391481-3f3d4c85-ef91-4663-a571-998750d7c074.png">

_Real_ Courier is a fairly readable font, but the MS commissioned _Courier New_ is so very light it is unusable.

Times New Roman renders smaller than many other fonts for a given font-size. While newspapers try to save paper, this is an accessibility problem and there is no "paper" to save on the web.

And I included some more recent MS examples, like Gulim, Cordia, and Iris ... like, **_wut?_**

For the record, my critique of MS fonts is only and exclusively directed at accessibility issues of the cited fonts. But it is useful to point out that many of the Google fonts are also anti-accessible. And I do give MS credit for commissioning Matthew Carter to add some good solid designs to MS's fonts.

I have an [informal paper evaluating accessible fonts](https://www.researchgate.net/publication/338149302_Evaluating_Fonts_Font_Family_Selection_for_Accessibility_Display_Readability) available for download. In it, I give generally positive comments for MS Fonts Verdana and Georgia, but negative comments for Arial and especially Arial Narrow. Several Google fonts also received criticism for poor accessibility.

Thank you,

Andy

**Comment by mraccess77** [comment]
I've also noticed some fonts that have uneven baselines which make the font seem to move or wave.  Others have letters like e and c that have very small openings making the letters more difficult to distinguish.  Too thick of fonts are also challenges for some people as well as fonts that are horizontally or vertically stretched.

This article may be of use as well as the Readability Group is working on some research into this area: 

https://medium.com/the-readability-group/a-guide-to-understanding-what-makes-a-typeface-accessible-and-how-to-make-informed-decisions-9e5c0b9040a0

**Reply by Myndex** [reply]
Hi Jonathan @mraccess77 

> I've also noticed some fonts that have uneven baselines which make the font seem to move or wave. Others have letters like e and c that have very small openings making the letters more difficult to distinguish. 

There are a lot of bad fonts out there. And there is literally no standardization. Something like uneven baselines could be bad hinting for instance, or bad kerning or tracking.

But also, the method that text is rendered to the screen has an impact. Some rasterizers do not consider kerning properly, resulting in a very uneven look that is harder to read.

And "small counters" is a common problem for badly hinted screen fonts. (A counter is the hole in the "e" for instance). MS had a font "Century Gothic" and the small counters make it unusable.

> Too thick of fonts are also challenges for some people as well as fonts that are horizontally or vertically stretched.

Yes, in the APCA guidelines on contrast for WCAG 3/Silver, smaller fonts are disallowed from using 800 or 900 weight.

And yes, font aspect ratio, tracking, kerning, leading (letter and line spacing) are all key parts of a good vs bad font.

> This article may be of use as well as the Readability Group is working on some research into this area:
> https://medium.com/the-readability-group/a-guide-to-understanding-what-makes-a-typeface-accessible-and-how-to-make-informed-decisions-9e5c0b9040a0

Indeed, I've had a few conversations with the author of that, Gareth Williams, on this subject. He had a research project going on at the BBC recently, not certain of the current status.

## CSS Related
I just made a larger post on this topic, discussing the need for accessible fallback generic fonts in CSS over at the CSS working draft:

https://github.com/w3c/csswg-drafts/issues/4910#issuecomment-930743523
